### PR TITLE
chore(deps): Remove Python 3.7, add numpy explicitly with the updated version

### PR DESCRIPTION
**Pull request by parthjaw**
## Description
I noticed a [Dependabot security alert](https://github.com/ginger-io/transformer-embeddings/security/dependabot/2) regarding a vulnerable Numpy version. Since it seems like Numpy gets installed as a transitive dependency of other packages, I added it explicitly to enforce versioning. This may seem sub-optimal since the right fix may be to update the packages that depend on Numpy rather than doing this. There are two reasons why I opted for this instead:

- Couldn't find the dependencies that would need to be updated
- On updating to the right Numpy version, we got test failures with Python 3.7. Since [3.7 EOL is imminent](https://endoflife.date/python), it made sense to make this change and also enforce Python versions >= 3.8. 

## Changes

- chore: Add Numpy explicitly to pyproject.toml. 
- chore: Remove 3.7 
- chore: Remove 3.7 from automated testing

We may need a follow-up PR to add support for Python 3.11. 

## Tests

- [x] Local.
- [ ] CI.

### CODE REVIEW

**Comment by parthjaw** [comment]
@setu4993 As expected, 3.7 is failing some of the tests. Curious if this change makes sense, or if there another preferred way to address this. Of course, if this change is acceptable, that would mean changing the tests/gh-action failing right now.

**Review by jaymeszm — APPROVED** [review]
Thanks, Parth! It seems like some of the packages that require numpy like pytorch doesn't specify the version required so forcing the numpy version from our end makes sense to me. 

Minor note on dropping support for 3.7 - let's update the README as well if we do.

File: noxfile.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from nox_poetry import Session, session
 
 
-@session(python=["3.7", "3.8", "3.9", "3.10"])
+@session(python=["3.8", "3.9", "3.10"])
```
Comment (setu4993) [inline noxfile.py:4]: Nice. We should also look at the GitHub Actions in case we need to update it to drop py37.

Comment (parthjaw) [reply noxfile.py:4]: Working on that right now! That would mean modifying the test-release.yml file, right? That's what I've done right now.

Comment (setu4993) [reply noxfile.py:4]: Correct.

File: pyproject.toml [hunk 2]
```diff
@@ -16,7 +16,8 @@ classifiers = [
 Changelog = "https://github.com/ginger-io/transformer-embeddings/releases"
 
 [tool.poetry.dependencies]
-python = "^3.7,<3.11"
+numpy = "^1.22.0"
```
[the remark this answers is not in the document]
Comment (parthjaw) [reply pyproject.toml]: This makes sense. Thanks! Basically rebuilt dependencies based on @bsweeney-gingerio's comment above!

File: pyproject.toml [hunk 3]
```diff
@@ -16,7 +16,7 @@ classifiers = [
 Changelog = "https://github.com/ginger-io/transformer-embeddings/releases"
 
 [tool.poetry.dependencies]
-python = "^3.7,<3.11"
+python = ">=3.8,<3.11"
```
Comment (setu4993) [inline pyproject.toml:19]: Nit: I prefer the `^` definitions more. [Docs.](https://python-poetry.org/docs/dependency-specification/)

```suggestion
python = "^3.8,<3.11"
```

**Review by setu4993 — APPROVED** [review]
Neat change. Thanks, Parth!

**Pull request merged**